project/src/main/puzzle/level/milestone.gd

```diff
@@ -49,3 +49,21 @@ func from_json_dict(json: Dictionary) -> void:
 	for key in json.keys():
 		if not key in ["type", "value"]:
 			set_meta(key, json.get(key))
+
+
+func to_json_dict() -> Dictionary:
+	var result := {
+		"type": Utils.enum_to_snake_case(MilestoneType, type),
+		"value": value,
+	}
+	for meta_key in get_meta_list():
+		result[meta_key] = get_meta(meta_key)
+	return result
+
+
+func is_default() -> bool:
```

Comment:
This is worth a comment.

Reply:
Added comment: `Milestones whose properties are set to their default values are omitted from our json output.`